capstone/backend/src/main/java/com/google/servlets/CreateUserServlet.java

```diff
@@ -0,0 +1,66 @@
+package com.google.servlets;
+
+import com.google.gson.Gson;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Arrays;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that adds a new user to the database. 
+ */
+@WebServlet("/api/v1/create-user")
+public class CreateUserServlet extends HttpServlet {
+  private static final long INVALID_USER_ID = -1;
+
+  private class RequestBody {
+    String email;
+    String userName;
+  }
+
+  private class ResponseBody {
+    long userId;
+    public ResponseBody(long userId) {
+      this.userId = userId;
+    }
+  }
+
+  /*
+   * This handles a POST request to "/create-user". A new user object is created and added to
+   * the Users table in the database.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson g = new Gson();
+    String reqString = ServletUtil.getRequestBody(request);
+    RequestBody requestBody = requestValidator(reqString);
+    if (requestBody == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid request syntax.");
+      return;
+    }
+    long userId = LibraryFunctions.createUser(requestBody.email.hashCode(), requestBody.userName, 
+        requestBody.email);
```

Comment:
Does this function return long? I see it returns void. Also don't use negative numbers to indicate exceptions. Just catch the exception when needed.

Reply:
I modified this in the PR #12 to catch a SpannerException.